src/AutoRest.CSharp.V3/AutoRest/Plugins/GeneratedCodeWorkspace.cs

```diff
@@ -128,5 +128,7 @@ public async Task<CSharpCompilation> GetCompilationAsync()
             Debug.Assert(compilation != null);
             return compilation;
         }
+
+        public static bool IsGeneratedDocument(Document document) => document.Folders.Contains(GeneratedFolder);
```

Comment:
Is it searching the file system each time to run this? Should we cache the result?

Reply:
Folders is an in-memory property of Roslyn `Document` type. Not IO here.